main/views.py

```diff
@@ -127,19 +140,99 @@ def get_object(self, queryset: Any | None = None) -> "UserType":
 
-    template_name = "main/about.html"
+    template_name = "main/pages/about.html"
 
 
 class TermsPageView(TemplateView):
     """View that renders the terms and conditions page."""
 
-    template_name = "main/terms.html"
+    template_name = "main/pages/terms.html"
+
+
+class SkillLevelsPageView(TemplateView):
+    """View that renders the terms and conditions page."""
+
+    template_name = "main/pages/skill-levels.html"
+
+
+class TrainingPageView(TemplateView):
+    """View that renders the terms and conditions page."""
 
+    template_name = "main/pages/training.html"
 
-class ContactPageView(TemplateView):
-    """View that renders the contact page."""
 
-    template_name = "main/contact.html"
+class RolesPageView(TemplateView):
+    """View that renders the terms and conditions page."""
+
+    template_name = "main/pages/roles.html"
+
+
+class GetInvolvedPageView(TemplateView):
+    """View that renders the terms and conditions page."""
+
+    template_name = "main/pages/get-involved.html"
+
+
+class EventsPageView(TemplateView):
+    """View that renders the events page."""
+
+    template_name = "main/pages/events.html"
+
+    def get_context_data(self, **kwargs: Mapping[str, object]) -> dict[str, object]:
+        """Add events from CSV to the template context."""
+        context = super().get_context_data(**kwargs)
+        csv_path = Path("data/events.csv")
+        events = []
+
+        if csv_path.exists():
+            with open(csv_path, newline="", encoding="utf-8") as csvfile:
+                reader = csv.DictReader(csvfile)
+                for row in reader:
+                    events.append(
+                        {
+                            "title": row.get("Title", ""),
+                            "start_date": row.get("Start Date", ""),
+                            "end_date": row.get("End Date", ""),
+                            "description": row.get("Description", ""),
+                            "contributors": row.get("Contributors", ""),
+                            "image": static(
+                                row.get("Image", "assets/img/blog/single/image.jpg")
+                            ),
+                        }
+                    )
+
+        events.sort(key=lambda e: e["start_date"], reverse=True)
+
+        context["events"] = events
+        return context
+
+
+class AccountOverviewPageView(TemplateView):
+    """View that renders the account overview page."""
+
+    template_name = "main/user_overview.html"
+
+
+class CompetenciesPageView(TemplateView):
+    """View that renders the competencies page."""
+
+    template_name = "main/pages/competencies.html"
+
+    def get_context_data(self, **kwargs: Mapping[str, object]) -> dict[str, object]:
+        """Add the competencies framework data to the template context."""
+        context = super().get_context_data(**kwargs)
+        json_path = Path("data/skills-competencies-framework.json")
```

Comment:
This should be changed to get the framework directly from the database instead of from the json file because there is a chance they will become out-of-sync.

For now this is fine though.